pkg/cli/plugin_cmd.go

```diff
@@ -5,20 +5,73 @@ package cli
 
 import (
 	"context"
+	"fmt"
 	"os"
 	"strconv"
 	"strings"
 
 	"github.com/spf13/cobra"
 
-	"github.com/vmware-tanzu/tanzu-cli/pkg/common"
+	configtypes "github.com/vmware-tanzu/tanzu-plugin-runtime/config/types"
 	"github.com/vmware-tanzu/tanzu-plugin-runtime/log"
+
+	"github.com/vmware-tanzu/tanzu-cli/pkg/common"
 )
 
+// K8s-targeted plugins command group shows up as top level commands and under
+// the kubernetes group. Any remap location that points to one should also map
+// to the other to be consistent.
+// This function returns the alternate location, if applicable.
+func alternateRemapLocation(p *PluginInfo, location string) string {
+	if p.Target == configtypes.TargetK8s {
+		cmdHierarchy := strings.Split(strings.TrimSpace(location), " ")
+		if len(cmdHierarchy) == 1 {
+			return fmt.Sprintf("kubernetes %s", cmdHierarchy[0])
+		}
+		if len(cmdHierarchy) == 2 && cmdHierarchy[0] == "kubernetes" {
+			return strings.Join(cmdHierarchy[1:], " ")
+		}
```

Comment:
Thanks for bring this up. As an experimental feature, the intention is to make available potentially useful mapping scenarios which includes:

- mapping a plugin to entirely replace that occupied by another
- placing the plugin's command tree under some other non-default "grouping" cmd 

But you noted in the plugin-runtime comment a case that I did not intend to support with this change, which is the ability to embed a plugin's command tree within that of another plugin's command tree. (think "tanzu space _builder cli add-plugin_", as an nonsensical example)
 
It is not because the mapping does not work, but it is more due to the "upper" plugin having no clue that this mapping has occurred (it's Usage will be incorrect) and that there are no known use cases that requires this.

So, I added as a topmost commit additional checks to prevent the remap if the parent Command is that which wraps a plugin.

Reply:
To are correct and I hadn’t thought through the implications of adding a plugin under another one.  Such a deeper remapping would only work (I think) under a core command’s command tree since the cli is aware of all of those. A plugin could for example add `tanzu context createtmc`. I’m only mentioning it as a possible future potential but I like that you blocked all that for now